code/core/src/manager/components/sidebar/SidebarBottom.tsx

```diff
@@ -214,18 +179,18 @@ export const SidebarBottomBase = ({
       }
     };
 
-    api.getChannel()?.on(TESTING_MODULE_CRASH_REPORT, onCrashReport);
-    api.getChannel()?.on(TESTING_MODULE_RUN_ALL_REQUEST, clearState);
-    api.getChannel()?.on(TESTING_MODULE_PROGRESS_REPORT, onProgressReport);
+    api.on(TESTING_MODULE_CRASH_REPORT, onCrashReport);
+    api.on(TESTING_MODULE_RUN_ALL_REQUEST, clearState);
+    api.on(TESTING_MODULE_PROGRESS_REPORT, onProgressReport);
 
     return () => {
-      api.getChannel()?.off(TESTING_MODULE_CRASH_REPORT, onCrashReport);
-      api.getChannel()?.off(TESTING_MODULE_PROGRESS_REPORT, onProgressReport);
-      api.getChannel()?.off(TESTING_MODULE_RUN_ALL_REQUEST, clearState);
+      api.off(TESTING_MODULE_CRASH_REPORT, onCrashReport);
+      api.off(TESTING_MODULE_PROGRESS_REPORT, onProgressReport);
+      api.off(TESTING_MODULE_RUN_ALL_REQUEST, clearState);
     };
```

Comment:
I'm considering moving this code into the state module as well.

Right now, every time `testProviders` changes (which happens a LOT due to that's where the state is), we unsubscribe and resubscribe.

It's technically possible to miss events during this short period.